src/test/app/TrustAndBalance_test.cpp

```diff
@@ -0,0 +1,512 @@
+
+        jrr = ledgerEntryState (env, gw, alice, "USD");
+        BEAST_EXPECT(
+            jrr[jss::node][sfBalance.fieldName][jss::value] == "0");
+        BEAST_EXPECT(
+            jrr[jss::node][sfHighLimit.fieldName][jss::value] == "700");
+        BEAST_EXPECT(
+            jrr[jss::node][sfHighLimit.fieldName][jss::issuer] == alice.human());
+        BEAST_EXPECT(
+            jrr[jss::node][sfHighLimit.fieldName][jss::currency] == "USD");
+        BEAST_EXPECT(
+            jrr[jss::node][sfLowLimit.fieldName][jss::value] == "0");
+        BEAST_EXPECT(
+            jrr[jss::node][sfLowLimit.fieldName][jss::issuer] == gw.human());
+        BEAST_EXPECT(
+            jrr[jss::node][sfLowLimit.fieldName][jss::currency] == "USD");
+
+        // set negative limit - expect failure
+        env (trust (alice, gw["USD"](-1)), ter(temBAD_LIMIT));
+
+        // set zero limit
+        env (trust (alice, gw["USD"](0)));
+
+        //ensure line is deleted
+        jrr = ledgerEntryState (env, gw, alice, "USD");
+        BEAST_EXPECT(jrr[jss::error] == "entryNotFound");
+
+        // TODO Check in both owner books.
+
+        // set another credit limit
+        env (trust (alice, bob["USD"](600)));
+
+        // set limit on other side
+        env (trust (bob, alice["USD"](500)));
+
+        // check the ledger state for the trust line
+        jrr = ledgerEntryState (env, alice, bob, "USD");
+        BEAST_EXPECT(
+            jrr[jss::node][sfBalance.fieldName][jss::value] == "0");
+        BEAST_EXPECT(
+            jrr[jss::node][sfHighLimit.fieldName][jss::value] == "500");
+        BEAST_EXPECT(
+            jrr[jss::node][sfHighLimit.fieldName][jss::issuer] == bob.human());
+        BEAST_EXPECT(
+            jrr[jss::node][sfHighLimit.fieldName][jss::currency] == "USD");
+        BEAST_EXPECT(
+            jrr[jss::node][sfLowLimit.fieldName][jss::value] == "600");
+        BEAST_EXPECT(
+            jrr[jss::node][sfLowLimit.fieldName][jss::issuer] == alice.human());
+        BEAST_EXPECT(
+            jrr[jss::node][sfLowLimit.fieldName][jss::currency] == "USD");
+    }
+
+    void
+    testDirectRipple ()
+    {
+        testcase ("Direct Payment, Ripple");
+        using namespace test::jtx;
+
+        Env env {*this};
+        Account alice {"alice"};
+        Account bob {"bob"};
+
+        env.fund (XRP(10000), alice, bob);
+        env.close();
+
+        env (trust (alice, bob["USD"](600)));
+        env (trust (bob, alice["USD"](700)));
+
+        // alice sends bob partial with alice as issuer
+        env (pay (alice, bob, alice["USD"](24)));
+        env.require (balance (bob, alice["USD"](24)));
+
+        // alice sends bob more with bob as issuer
+        env (pay (alice, bob, bob["USD"](33)));
+        env.require (balance (bob, alice["USD"](57)));
+
+        // bob sends back more than sent
+        env (pay (bob, alice, bob["USD"](90)));
+        env.require (balance (bob, alice["USD"](-33)));
```

Comment:
I found it useful to add
```c++
env.require (balance (alice, bob["USD"](33)));
```
to show the symmetry of the balances. 

Not important to add, just wanted to highlight what was useful for a novice ripple coder.